pytensor/tensor/math.py

```diff
@@ -622,7 +622,36 @@ def ge(a, b):
 
 @scalar_elemwise
 def eq(a, b):
-    """a == b"""
+    """Element-wise equality comparison.
+
+    Computes a tensor of 0s and 1s where 1 indicates the corresponding
+    elements of a and b are equal.
+
+    Parameters
+    ----------
+    a, b : tensor_like
+        Input tensors of same shape.
+
+    Returns
+    -------
+    tensor
+        Boolean tensor of same shape as inputs, with 1s where corresponding
+        elements are equal, 0s otherwise.
+
+    Examples
+    --------
+    >>> import pytensor.tensor as pt
+    >>> import numpy as np
+    >>> x = pt.vector()
+    >>> y = pt.vector()
+    >>> f = pt.function([x, y], pt.eq(x, y))
+    >>> f([1, 2, 3], [1, 4, 3])
+    array([ True, False,  True])
+
+    Notes
+    -----
+    This function supports the Python syntax `a == b` when used with PyTensor tensors.
```

Comment:
Weird to say this, a == b is explicitly not supported, I guess an "equivalent" is missing